modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/FilterParserImplTest.java

```diff
@@ -77,6 +77,51 @@ public void testParseNonexistingField() {
 			filterString);
 	}
 
+	@Test
+	public void testParseSuboperationWithDateAndDuration()
+		throws ExpressionVisitException {
+
+		Expression expression = _filterParserImpl.parse(
+			"dateTimeExternal ge (2022-12-28T23:00:00.000Z sub duration'PT24H')");
+
+		Assert.assertNotNull(expression);
```

Comment:
You don't neet to test the assertNotNull, the error in that case is a NullPointer, the same as a assertNotNull.

Reply:
ok